### DISCUSSION TITLE
Tautomer generator based on the InChI

### DISCUSSION THREAD

**Discussion by mrhso** [rdkit/rdkit#5305] [Ideas]
In order to use scoring system to determine canonical tautomer without changing the standard InChI for molecules, it is worth considering to introduce a tautomer generator based on the InChI.
BTW, in CDK the class for this is `InChITautomerGenerator`.

**Comment by bp-kelley** [comment]
The reference paper for this method is here:

https://pubs.acs.org/doi/10.1021/ci1001179

**Comment by mrhso** [comment]
For non-standard InChI (especially for fixed H), InChI strings can be used as input.
